pallets/proofs-dealer/src/lib.rs

```diff
@@ -752,6 +774,30 @@ pub mod pallet {
             // This TX is free since is a sudo-only transaction used for testing.
             Ok(Pays::No.into())
         }
+
+        #[pallet::call_index(4)]
+        #[pallet::weight(Weight::zero())]
+        pub fn priority_challenge(
+            origin: OriginFor<T>,
+            key: KeyFor<T>,
+            should_remove_key: bool,
+        ) -> DispatchResultWithPostInfo {
+            // Check that the extrinsic was executed by the custom origin.
+            T::PriorityChallengeOrigin::ensure_origin(origin.clone())?;
+
+            let raw_origin: RawOrigin<T::AccountId> =
+                origin.into().map_err(|_| DispatchError::BadOrigin)?;
+
+            let who = match &raw_origin {
+                RawOrigin::Signed(account) => Some(account.clone()),
+                RawOrigin::Root | RawOrigin::None => None,
+            };
```

Comment:
We can rely on the `into_signer` function to return an optional account id from an origin.

```suggestion
use frame_support::traits::OriginTrait;

// Check that the extrinsic was executed by the custom origin.
T::ChallengeOrigin::ensure_origin(origin.clone())?;

let who = origin.into_signer();
```